### PR TITLE
[Autorest.Android]: Merge autorest_v4_android_v2 to v4 branch

### PR DESCRIPTION
**Pull request by JianpingChen**

### CODE REVIEW

File: fluent-tests/prepare-tests.bat [hunk 1]
```diff
@@ -12,7 +12,7 @@ RMDIR /S /Q "src\main\java\com\azure\mgmtlitetest"
 
 SET AUTOREST_CORE_VERSION=3.1.3
 SET MODELERFOUR_ARGUMENTS=--pipeline.modelerfour.additional-checks=false --pipeline.modelerfour.lenient-model-deduplication=true
-SET COMMON_ARGUMENTS=--java --use=../ --java.output-folder=./ %MODELERFOUR_ARGUMENTS% --azure-arm --java.license-header=MICROSOFT_MIT_SMALL
+SET COMMON_ARGUMENTS=--java --use=../ --java.output-folder=./ %MODELERFOUR_ARGUMENTS% --azure-arm --java.license-header=MICROSOFT_MIT_SMALL --use=@autorest/modelerfour@4.18.0
```
Comment (weidongxu-microsoft) [inline fluent-tests/prepare-tests.bat:15]: Why 4.18.0?

If you really want that version, I could try it in v4. It should be in readme of fluentnamer.

File: fluent-tests/report.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Java Codegen Report
```
Comment (weidongxu-microsoft) [inline fluent-tests/report.md:1]: this file does not matter, you do not need to merge or else, you can just delete it if conflict.

File: readme.md [hunk 3]
```diff
@@ -743,14 +743,18 @@ AutoRest needs the below config to pick this up as a plug-in - see https://githu
 
 #### Javagen
 
-``` yaml $(java) && !$(fluent)
+``` yaml $(java) && !$(fluent) && !$(android)
 use: $(this-folder)/javagen
 ```
 
-``` yaml $(java) && $(fluent)
+``` yaml $(java) && $(fluent) && && !$(android)
```
Comment (weidongxu-microsoft) [inline readme.md:750]: 2 "&&"?

Comment (weidongxu-microsoft) [reply readme.md:750]: fluentgen seems not happen in fluent - ci. Not sure whether this is the cause.

File: fluent-tests/prepare-tests.bat (hunk 1 above)
Comment (weidongxu-microsoft) [reply fluent-tests/prepare-tests.bat:15]: I've upgraded to 4.18.1 https://github.com/Azure/autorest.java/pull/1013

File: androidgen/pom.xml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+<modelVersion>4.0.0</modelVersion>
+<parent>
+    <groupId>com.azure.tools</groupId>
+    <artifactId>azure-autorest-parent</artifactId>
+    <version>1.0.0-beta.1</version>
+    <relativePath>../pom.xml</relativePath>
+</parent>
+<groupId>com.azure.tools</groupId>
+<artifactId>androidgen</artifactId>
+<version>1.0.0-SNAPSHOT</version>
```
Comment (JianpingChen) [inline androidgen/pom.xml:14]: Should this also be "1.0.0-beta.1" as other plugins?

**Review by jianghaolu — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

**Pull request merged**